openedx/features/edly/utils.py

```diff
@@ -0,0 +1,255 @@
+
+
+def notify_students_about_xblock_changes(xblock, publish, old_content):
+    """
+    This function is responsible for calling the related function by checking the xblock category.
+
+    :param xblock: Block of courses which has been published.
+    :param publish: Param to check if xblock is going to be published.
+    :param old_content: Old data of xblock before updating.
+    """
+    if (publish == 'make_public' and xblock.category in COURSE_OUTLINE_CATEGORIES
+        and not xblock.visible_to_staff_only):
+        _handle_section_publish(xblock)
+    elif xblock.category == 'course_info' and xblock.location.block_id == 'handouts':
+        _handle_handout_changes(xblock, old_content)
+
+
+def get_email_params(xblock):
+    """
+    This function is responsible for generating the email_params which we are
+    going to send to students for any changes in course.
+
+    :param xblock: xblock which is modified/created
+    :return: Dict containing the data for email.
+    """
+    email_params = {}
+    course = get_course_by_id(xblock.location.course_key)
+    email_params['course_url'] = _get_course_url(xblock.location.course_key)
+    email_params['course_name'] = course.display_name_with_default
+    email_params['display_name'] = xblock.display_name
+    email_params['site_name'] = configuration_helpers.get_value(
+        'SITE_NAME',
+        settings.SITE_NAME
+    )
+    return email_params
+
+
+def _get_course_url(course_key):
+    return '{}/courses/{}'.format(settings.LMS_ROOT_URL, course_key)
+
+
+def _handle_section_publish(xblock):
+    """
+    This function will send email to the enrolled students in the case
+    of any outline changes like section, subsection, unit publish.
+
+    :param xblock: xblock which is modified/created
+    """
+    email_params = get_email_params(xblock)
+    students = get_course_enrollments(xblock.location.course_key)
+
+    if xblock.category == 'vertical':
+        published_unit_url = _get_published_unit_url(xblock)
+        email_params['change_type'] = 'Unit'
+        email_params['change_url'] = published_unit_url
+    elif xblock.category == 'sequential':
+        subsection_url = '{lms_base_url}/courses/{course_id}/course/#{subsection_usage_key}'.format(
+            lms_base_url=settings.LMS_ROOT_URL,
+            course_id=xblock.location.course_key,
+            subsection_usage_key=xblock.location
+        )
+        email_params['change_url'] = subsection_url
+        email_params['change_type'] = 'Sub Section'
+    else:
+        section_url = '{lms_base_url}/courses/{course_id}/course/#{section_usage_key}'.format(
+            lms_base_url=settings.LMS_ROOT_URL,
+            course_id=xblock.location.course_key,
+            section_usage_key=xblock.location
+        )
+        email_params['change_type'] = 'Section'
+        email_params['change_url'] = section_url
+    send_bulk_mail_to_students.delay(students, email_params, 'outline_changes')
```

Comment:
On publishing `sub seciton` I got this 
```

------- EMAIL -------
To: edx@example.com
From: 
Subject: Your course hassan_demo_2 on localhost:8001 has been updated
Body:
Dear 

Sub Section: New Hello has been updated in hassan_demo_2.

Your can see the Sub Section changes here: http://localhost:18000/courses/course-v1:ucp+CS1034+2014_T2/course/#block-v1:ucp+CS1034+2014_T2+type@sequential+block@4d286546ec12486b812639767263c127

----
Thanks,
LUMS ELM training team
-------  END  -------
```

Kindly replace the line `Your can see ....` to `You can view ....`

Reply:
I think See is a better option here. You can check the difference b/w view and see [here](https://wikidiff.com/view/see)

Reply:
Done